src/hypofuzz/hypofuzz.py

```diff
@@ -289,32 +286,23 @@ def fuzz(self) -> None:
         # TODO: make this aware of test runtime, so it adapts for behaviors-per-second
         #       rather than behaviors-per-input.
 
-        resort = False
-        for count in itertools.count():
-            if count % 20 == 0:
-                resort = True
-                i = self.random.randrange(len(self.targets))
-            else:
-                i = 0
-            target = self.targets[i]
-            target.run_one()
-            if target.has_found_failure:
-                print(f"found failing example for {target.nodeid}")
-                self.targets.pop(i)
-
-            if self.targets and (
-                resort
-                or (
-                    len(self.targets) > 1
-                    and self.targets.key(self.targets[0])
-                    > self.targets.key(self.targets[1])
-                )
-            ):
-                # pay our log-n cost to keep the list sorted
-                self.targets.add(self.targets.pop(0))
+        while True:
+            for target in self.targets:
+                if target.has_found_failure:
+                    print(f"found failing example for {target.nodeid}")
+                    self.targets.remove(target)
 
             if not self.targets:
-                return
+                break
+
+            # choose the next target to fuzz with probability equal to the softmax
+            # of its estimator.
+            estimators = [behaviors_per_second(target) for target in self.targets]
+            estimators = softmax(estimators)
+            target = self.random.choices(self.targets, weights=estimators, k=1)[0]
+
+            for _ in range(100):
+                target.run_one()
```

Comment:
todo comment for a future PR: we could also scale this number up if we estimate a long time-to-bug or large gap to next-best option, to reduce switching overhead.  Similarly we could account for fixture costs in future.